### PR TITLE
HAAR-1340 Case Note Performance Improvement - Remove SQL call to group case note types, use in memory grouping

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 1]
```diff
@@ -444,7 +444,7 @@ public List<Assessment> getAssessments(@PathVariable("bookingId") @Parameter(des
         return inmateService.getAssessments(bookingId);
     }
 
-    @Operation(summary = "Offender case notes.", description = "Offender case notes.")
+    @Operation(summary = "Offender case notes.", description = "Offender case notes.", hidden = true)
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:447]: These endpoint shouldn't be used by clients (only case note service)

Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:447]: Maybe add  a new role that only the case notes service has to restrict access?

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:447]: That would make sense - obs not in this PR

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**